crates/autopilot/src/solvable_orders.rs

```diff
@@ -141,6 +143,43 @@ impl SolvableOrdersCache {
             protocol_fees,
             cow_amm_registry,
         });
+
+        let solvable_orders_cache_clone = self_.clone();
+        banned_users.spawn_maintenance_task(move |cache| {
```

Comment:
This adds a lot of complexity imo (it leaks a lot of implementation detail of banned users, the fact that there is maintenance, that there is a cache, how it is structured, etc into the solvable orders cache). I assume passing a way to retrieve the current auction into the banned user component was considered but isn't possible?

Why can't we simply update the entire map of tracked users? Are we worried about the cache becoming too big?

Reply:
Yeah, I don't like it either. I initially was thinking about LRU cache to avoid a lot of entries, but @MartinquaXD  suggested keeping only long-lasting limit order participants as a more straightforward and concise approach. However, due to circular dependencies, this adds a lot of complexity. Maybe LRU should be reconsidered then.

Reply:
> Why can't we simply update the entire map of tracked users? Are we worried about the cache becoming too big?

Yeah, that was the concern my suggestion intended to address.

Reply:
Given that we already have a way more serious memory leak which we still manage to address without facing terrible consquences I guess having an unbounded cache for addresses is not the worst thing. This is not the hill I want to die on so if you both prefer the low complexity solution that never evicts from the cache feel free to go with that. 👌

Reply:
I am still a bit worried about the unbounded cache, which we have to fetch info for each address in the background task. We have around 5k unique addresses accumulated during the week. Making 5k calls on each maintenance task iteration seems overkill, doesn't it?

Reply:
We should still cache the status of the user so we should end up doing ~5k requests per hour, no?
That is perfectly reasonable IMO.

Reply:
Ok, updated with the unbounded cache.